robotics/mainHandler.go

```diff
@@ -11,23 +11,29 @@ import (
 	"gobot.io/x/gobot/platforms/raspi"
 )
 
-var adaptor = raspi.NewAdaptor()
+var raspiAdaptor = raspi.NewAdaptor()
 var robot *gobot.Robot
 var ticker *time.Ticker
 
+func initRobot() {
+	robot = gobot.NewRobot("raspiBot",
+		[]gobot.Connection{raspiAdaptor},
+		StartListening,
+	)
+}
+
 func TurnOnCar(w http.ResponseWriter, req *http.Request) {
 	fmt.Fprint(w, "Starting car!\n")
-	//StartMotors(byte(255), adaptor)
-	db.StartCar()
 	fmt.Printf("%+v\n", robot)
+
+	db.StartCar()
 	if robot != nil {
 		if !robot.Running() {
-			robot = gobot.NewRobot("raspiBot",
-				[]gobot.Connection{adaptor},
-				StartListening,
-			)
 			robot.Start()
 		}
+	} else {
+		initRobot()
+		TurnOnCar(w, req)
```

Comment:
I guess it's because robot needs to start, and the init function does not call the "Start()" function, so once the object is initialized, the first conditional is going to be true.